Replace Locate selection limit with confirmation dialog

AbeJellinek opened the pull request:
Also corrects an inconsistency between Enter key behavior, which had a max of 19, and Locate menu behavior, which had a max of 20.

Closes #2704.

AbeJellinek committed:
Fix Scaffold load error when fontSize pref is set

AbeJellinek committed:
Scaffold templates: Pull URL from doc

Link URLs could lead to redirects.

dstillman committed:
Try to open external PDF reader to page number for "Show on Page"

PDF-handling code from `ZoteroPane.viewAttachment()` predates
`Zotero.OpenPDF` and didn't handle page numbers for external PDF
readers, so when called from "Show on Page" in a note (via
`ZoteroPane.viewPDF()`) it just launched the file itself in the
specified handler or via the OS.

For now, change to use `Zotero.OpenPDF.openToPage()` when there's a page
number. Later we should use `Zotero.OpenPDF` for everything, but that
code is more complicated than just using `launchFile()`, so we'll need
to do some testing (particularly on Linux) to see what's best.

(Note that "Show on Page" with an external reader doesn't make a ton of
sense, since you can't actually see the annotation, but opening the
right page is better than nothing.)

Fixes #2612

dstillman committed:
Update translators, styles, and CSL locales

dstillman committed:
Update locales from Transifex

mrtcode committed:
Avoid error interrupting annotation extraction from multiple items

dstillman committed:
Update note-editor submodule

dstillman committed:
Update version

dstillman committed:
Feed import: Don't fail on OPML entry with no `title` or `text`

https://forums.zotero.org/discussion/96841/impossible-dimporter-ompl-rss

dstillman committed:
Add prefs for timeouts for downloading PDF via browser

1 second isn't enough for some users, so try 1.5, and allow for easier
debugging via a pref.

https://forums.zotero.org/discussion/96431/i-have-access-to-pdf-but-zotero-cannot-automatically-add-it
https://forums.zotero.org/discussion/97277/unable-to-download-papers-using-zotero-chrome-extension

dstillman committed:
Update note-editor submodule

dstillman committed:
citeproc-js: Add space after left margin in plain-text mode

This adds a space after the the number in some numeric styles ("[1]" in
IEEE style, "1." in Nature) when pasting bibliography entries into a
plain-text editor.

Some styles (ACS, AMA, Vancouver) will need to be updated to remove
their hard-coded spaces to avoid double spaces after this change.

Patch from @zepinglee

Fixes #2633

dstillman committed:
Update locales from Transifex

dstillman committed:
Update global schema

Includes the Preprint archiveID/number mapping for #2481

dstillman committed:
Fix support tests after 28ed3e34b

dstillman committed:
Fix detection of some emoji set as colored tags

Emoji such as ⭐️ that began as text-based Unicode characters and depend
on Variation Selector-16 to display the emoji presentation weren't being
properly detected.

mrtcode committed:
Fix library tab tooltip always staying on top

adomasven committed:
Fix cookieSandbox not overriding user agent if no cookies for domain.

Potential fix for #2635

adomasven committed:
Improve downloadPDFViaBrowser handlers and logging

adomasven committed:
Fix a MIMETypeHandler.removeHandler() bug

adomasven committed:
Log full hidden browser count after its creation

mrtcode committed:
Add support for PDF page deletion and rotation (#2595)

* Add support for PDF page deletion and rotation
Fixes #2561

dstillman committed:
Update locales from Transifex and merge new English strings

AbeJellinek committed:
Fix sorting by attachment (#2586)

dstillman committed:
Update styles submodule

dstillman committed:
Fix Emoji detection

And move to Zotero.Utilities.Internal.isOnlyEmoji()

Fixes #2643

dstillman committed:
Properly strip ZWJ in `isOnlyEmoji()`

dstillman committed:
Fix parsing of Chinese and other non-ASCII month names

https://forums.zotero.org/discussion/97683/bug-locale-issues-concerning-zotero-and-zotero-connector

dstillman committed:
Use String.fromCharCode() to load Chinese characters in strToDate() test

Follow-up to dc045250a6

AbeJellinek committed:
Fix error if New Collection prompt is canceled

adomasven committed:
Fix force-click during drag sometimes causing collections to switch

Closes #2652

adomasven committed:
Fix alt-arrow not working on macOS when editing collection names.

Fixes #2651

adomasven committed:
Limit the force-click during drag fix to macOS (4fb0661b)

mrtcode committed:
Update pdf-reader and note-editor, use specific prebuilds, tweak code

dstillman committed:
Update translators

dstillman committed:
Update locales from Transifex

dstillman committed:
Update utilities submodule and date-parsing tests

Includes strToDate() fixes

dstillman committed:
Update version

mrtcode committed:
Implement smart notes switching in contextPane (#2671)

Fixes #2650

dstillman committed:
Update pdf-reader submodule

mrtcode committed:
Show parent item title for child notes in context pane
Fixes #2681

mrtcode committed:
Adjust page rotation labels in PDF reader
Fixes #2669

dstillman committed:
Update global schema and utilities for CSL 1.0.2

mrtcode committed:
Fix disabled "Add to Note" when a child note is open
Fixes #2687

tnajdek committed:
Don't try to remove temporary dir after import

Previously we had a structure that also created folders in temp dir,
that has since changed but code in cleanup phase to remove containing
dir has been accidentally left unremoved.

tnajdek committed:
Mendeley import: detect win close, better feedback

* Detect wizard cancel/close and interrupt import. This will still
  have to wait for current fetch (file or metadata) to complete but will
  then advance to the cleanup stage
* Advance progress bar during metadata fetch
* Add some extra logging

tnajdek committed:
Fix a bug in regex extracting fields to "extra"

Because regex is built using a template string, \s* is actually escaped
into s*, i.e. literal "s" appearing 0 or more times. In most cases this
would mean that output can have spacing slightly off. In extreme case,
when identifier starts with letter "s", this could this could lead to
identifier being stored incorrectly.

Also adjusted tests to be more strict and mock data to cover this case.

tnajdek committed:
Lint code related to Mendeley importer

tnajdek committed:
Improve log output for multiple Mendeley imports

Helps with debugging cases where new import is started while previous
one is still being cancelled

dstillman committed:
Validate fields in ItemTree::getSortFields()

To avoid startup hang if a plugin does something bad:

https://forums.zotero.org/discussion/comment/411843/#Comment_411843

Fixes #2692

dstillman committed:
Fix test failure after a8ed30ce8

We added a `toggleSort()` call on the title, so then when called again
it reverses the direction. To avoid, sort by creator and then title in
the second test.

dstillman committed:
Add CSL 1.0.2 locators and switch to CSL locales for localization

And move Page to middle of list, while still selecting by default

https://forums.zotero.org/discussion/comment/396602/#Comment_396602

dstillman committed:
Update locales from Transifex

dstillman committed:
Update pdf-reader and pdf-worker submodules

AbeJellinek committed:
`recursiveCollections`: Remove from all subcollections on delete (#2689)

dstillman committed:
Merge pull request #2693 from tnajdek/mendeley-import-improve-feedback

Multiple improvements to Mendeley importer

dstillman committed:
Tweak confirmation message for recursive collection-item removal

I guess the "its" is technically a little clearer, but it's a bit
awkward, and I think it's reasonably implied.

dstillman committed:
Merge new English strings

dstillman committed:
Add localized string for `timestamp` CSL locator

Not included in the CSL locales:

https://forums.zotero.org/discussion/comment/412060/#Comment_412060

dstillman committed:
Merge new English strings

dstillman committed:
CookieSandbox: Fix error setting request cookies

Regression from 6af3b0276a, showing up as "trackedBy is undefined" error

https://forums.zotero.org/discussion/98322/no-snapshot-or-pdf-autosaved-using-zotero-connector-on-safari

dstillman committed:
Increase `downloadPDFViaBrowser.onLoadTimeout` to 3000

Apparently necessary for some people to get PDFs on ScienceDirect. We
should come up with a better way of detecting the page load if we can,
so that we can avoid this long of a timeout for people who just will
never have access to ScienceDirect PDFs.

https://forums.zotero.org/discussion/comment/411056/#Comment_411056

Addresses #2698

dstillman committed:
Fix sorting by Item Type and feed item itemID

Regression from a8ed30ce803

dstillman committed:
Fix sorting of attachments and notes columns

Regression from a8ed30ce803

dstillman committed:
Sort empty rows last for Date/Year columns when descending

Date columns start in descending order, so put empty rows at end

Also remove a seemingly unnecessary exception for Title when not the
primary sort and just always sort it with empty rows first

dstillman committed:
Revert "Validate fields in ItemTree::getSortFields()"

This reverts commit a8ed30ce803a9c5b3b5c04d50f6bd6b94faeee86 and related
commits.

We'll address breakage from invalid sort fields another way, without
inconveniencing plugin authors.

https://groups.google.com/g/zotero-dev/c/kc0-C6-SA74/m/bhHniGceAQAJ

dstillman committed:
Better fix for errors from invalid sort fields

Just catch the error from `ItemTree::sort()` and clear the
secondary-sort and fallback-sort prefs so that sorting works on the next
attempt.

Replacement for a8ed30ce803a9c5b3b5c04d50f6bd6b94faeee86

https://groups.google.com/g/zotero-dev/c/kc0-C6-SA74/m/bhHniGceAQAJ

dstillman committed:
Fix test failure after 26a5e2900a

mrtcode committed:
Include quotation marks for highlights by default, but allow to disable
Fixes #2697

mrtcode committed:
Don't include quotes when {{highlight}} is placed in a blockquote
Fixes #2697

mrtcode committed:
Prevent delayed context pane expansion and pdf-reader rerender/resize when switching from library tab

AbeJellinek committed:
Prevent deleting My Publications

Previously showed a blank confirmation dialog and logged an error.

mrtcode committed:
Use Quick Copy (Markdown) for direct plain-text copying from note-editor
Fixes #2505

AbeJellinek committed:
Replace hardcoded Locate selection limit with pref

Also corrects an inconsistency between Enter key behavior, which had a
max of 19, and Locate menu behavior, which had a max of 20.

dstillman commented:
https://forums.zotero.org/discussion/comment/412559/#Comment_412559

Just prompting above 20 actually kind of seems better. We should probably just do that. (Sorry, I was too scared to actually try this in a browser to see what they do…)

AbeJellinek committed:
Prompt instead of using pref

Prompt shows for Locate options and item tree activations and lists the
number of items to be opened.

AbeJellinek commented:
OK, replaced with a dialog. Let me know how that looks.

AbeJellinek committed:
genericTitle -> genericAction to match argument name
